forbidigo/forbidigo_test.go

```diff
@@ -19,6 +19,18 @@ func foo() {
 }`, "use of `fmt.Printf` forbidden by pattern `fmt\\.Printf` at testing.go:5:2")
 	})
 
+	t.Run("it finds forbidden, renamed identifiers", func(t *testing.T) {
+		linter, _ := NewLinter([]string{`fmt\.Printf`})
+		expectIssues(t, linter, `
+package bar
+
+import renamed "fmt"
+
+func foo() {
+	renamed.Printf("here i am")
+}` /* only detected inside golanci-lint */)
```

Comment:
Would it be possible to construct a working test for this in this file?  I assume we'd need to gather the TypesInfo somehow.

Reply:
That's the problem - I don't know how.

I agree that it would be nice to run "go test" here and get full coverage. On the other other, writing the test cases in https://github.com/ashanbrown/forbidigo/blob/master/pkg/analyzer/testdata/forbidden.go is so much simpler and will ensure coverage for PRs or when running "go test ./..." in the root that I am not sure whether it is worth the effort.

Reply:
It looks to me like it could be possible to construct a sufficient TypesInfo object to get the desired behavior in a test just by filling out what's in https://pkg.go.dev/go/types#Info.  I don't think we'd need to actually parse a package to do this.

Reply:
I came up with a solution where the tests in this directory are more like the code in main.go: they call `packages.Load` for some directory that I am setting up, then call a new `Run` function which takes packages and contains the code that previously was in `main.go` to iterate over those.

I found that with `packages.NeedTypesInfo | packages.NeedDeps` to the `Mode`, I get the necessary additional information about imports, so this new functionality will also be available in the stand-alone linter.

I need to clean this up a bit more and also port the other PR, then I'll push an update.

Reply:
I pushed to both PRs.